clang/lib/Sema/SemaTemplateInstantiate.cpp

```diff
@@ -4206,18 +4206,14 @@ Sema::InstantiateClassMembers(SourceLocation PointOfInstantiation,
         if (Function->hasAttr<ExcludeFromExplicitInstantiationAttr>())
           continue;
 
-        MemberSpecializationInfo *MSInfo =
-            Function->getMemberSpecializationInfo();
-        assert(MSInfo && "No member specialization information?");
-        if (MSInfo->getTemplateSpecializationKind()
-                                                 == TSK_ExplicitSpecialization)
+        TemplateSpecializationKind PrevTSK =
+            Function->getTemplateSpecializationKind();
+        if (PrevTSK == TSK_ExplicitSpecialization)
           continue;
```

Comment:
IIUC the new behavior handles `DependentFunctionTemplateSpecializationInfo` as well, right?

Reply:
I don't think that would matter, unless the assert above was wrong, as the `DependentFunctionTemplateSpecializationInfo` case is handled in the fallback path, when `MSInfo` is null.

Reply:
Yeah, I thought the bug was that somewhere we failed to handle `DependentFunctionTemplateSpecializationInfo`s, but after stepping into the codes, I realized that it is due to the latest TSK stored in `FunctionTemplateSpecializationInfo` rather than in the MSInfo. (We didn't update the TSK in MSInfo in CheckFunctionTemplateSpecialization)